metricbeat/module/haproxy/haproxy.go

```diff
@@ -0,0 +1,243 @@
+	Maxsock                    string `mapstructure:"Maxsock"`
+	Maxconn                    string `mapstructure:"Maxconn"`
+	HardMaxconn                string `mapstructure:"Hard_maxconn"`
+	CurrConns                  string `mapstructure:"CurrConns"`
+	CumConns                   string `mapstructure:"CumConns"`
+	CumReq                     string `mapstructure:"CumReq"`
+	MaxSslConns                string `mapstructure:"MaxSslConns"`
+	CurrSslConns               string `mapstructure:"CurrSslConns"`
+	CumSslConns                string `mapstructure:"CumSslConns"`
+	Maxpipes                   string `mapstructure:"Maxpipes"`
+	PipesUsed                  string `mapstructure:"PipesUsed"`
+	PipesFree                  string `mapstructure:"PipesFree"`
+	ConnRate                   string `mapstructure:"ConnRate"`
+	ConnRateLimit              string `mapstructure:"ConnRateLimit"`
+	MaxConnRate                string `mapstructure:"MaxConnRate"`
+	SessRate                   string `mapstructure:"SessRate"`
+	SessRateLimit              string `mapstructure:"SessRateLimit"`
+	MaxSessRate                string `mapstructure:"MaxSessRate"`
+	SslRate                    string `mapstructure:"SslRate"`
+	SslRateLimit               string `mapstructure:"SslRateLimit"`
+	MaxSslRate                 string `mapstructure:"MaxSslRate"`
+	SslFrontendKeyRate         string `mapstructure:"SslFrontendKeyRate"`
+	SslFrontendMaxKeyRate      string `mapstructure:"SslFrontendMaxKeyRate"`
+	SslFrontendSessionReusePct string `mapstructure:"SslFrontendSessionReuse_pct"`
+	SslBackendKeyRate          string `mapstructure:"SslBackendKeyRate"`
+	SslBackendMaxKeyRate       string `mapstructure:"SslBackendMaxKeyRate"`
+	SslCacheLookups            string `mapstructure:"SslCacheLookups"`
+	SslCacheMisses             string `mapstructure:"SslCacheMisses"`
+	CompressBpsIn              string `mapstructure:"CompressBpsIn"`
+	CompressBpsOut             string `mapstructure:"CompressBpsOut"`
+	CompressBpsRateLim         string `mapstructure:"CompressBpsRateLim"`
+	ZlibMemUsage               string `mapstructure:"ZlibMemUsage"`
+	MaxZlibMemUsage            string `mapstructure:"MaxZlibMemUsage"`
+	Tasks                      string `mapstructure:"Tasks"`
+	RunQueue                   string `mapstructure:"Run_queue"`
+	IdlePct                    string `mapstructure:"Idle_pct"`
+	Node                       string `mapstructure:"Node"`
+	Description                string `mapstructure:"Description"`
+}
+
+// Client is an instance of the HAProxy client
+type Client struct {
+	Address     string
+	ProtoScheme string
+}
+
+// NewHaproxyClient returns a new instance of HaproxyClient
+func NewHaproxyClient(address string) (*Client, error) {
+	parts := strings.Split(address, "://")
+	if len(parts) != 2 {
+		return nil, errors.New("Must have protocol scheme and address!")
+	}
+
+	if parts[0] != "tcp" && parts[0] != "unix" {
+		return nil, errors.New("Invalid Protocol Scheme!")
```

Comment:
Error strings should not be capitalized. https://github.com/golang/go/wiki/CodeReviewComments#error-strings